### ISSUE TITLE
Danger 6.3.0 fails with Octokit::NotFound on Bitrise CI

### ISSUE THREAD

**Issue by koral--** [danger/danger#1196]
# Report

## What did you do?

Run `bundle "exec" "danger" "--fail-on-errors=true"` invoked from official Bitrise Danger step.

## What did you expect to happen?

Running your Dangerfile without it crashing.
On 6.2.2 it is working correctly

## What happened instead?

```
Running danger
$ bundle "exec" "danger" "--fail-on-errors=true"
/usr/local/lib/ruby/gems/2.7.0/gems/git-1.6.0/lib/git/lib.rb:1054: warning: Using the last argument as keyword parameters is deprecated
bundler: failed to load command: danger (/usr/local/bin/danger)
Octokit::NotFound: GET https://api.github.com/repos/DroidsOnRoids/<stripped>.git/pulls/189: 404 - Not Found // See: https://developer.github.com/v3/pulls/#get-a-single-pull-request
  /usr/local/lib/ruby/gems/2.7.0/gems/octokit-4.16.0/lib/octokit/response/raise_error.rb:16:in `on_complete'
  /usr/local/lib/ruby/gems/2.7.0/gems/faraday-0.17.3/lib/faraday/response.rb:9:in `block in call'
  /usr/local/lib/ruby/gems/2.7.0/gems/faraday-0.17.3/lib/faraday/response.rb:61:in `on_complete'
  /usr/local/lib/ruby/gems/2.7.0/gems/faraday-0.17.3/lib/faraday/response.rb:8:in `call'
  /usr/local/lib/ruby/gems/2.7.0/gems/octokit-4.16.0/lib/octokit/middleware/follow_redirects.rb:73:in `perform_with_redirection'
  /usr/local/lib/ruby/gems/2.7.0/gems/octokit-4.16.0/lib/octokit/middleware/follow_redirects.rb:61:in `call'
  /usr/local/lib/ruby/gems/2.7.0/gems/faraday-0.17.3/lib/faraday/request/retry.rb:131:in `call'
  /usr/local/lib/ruby/gems/2.7.0/gems/faraday-0.17.3/lib/faraday/rack_builder.rb:143:in `build_response'
  /usr/local/lib/ruby/gems/2.7.0/gems/faraday-0.17.3/lib/faraday/connection.rb:387:in `run_request'
  /usr/local/lib/ruby/gems/2.7.0/gems/faraday-0.17.3/lib/faraday/connection.rb:138:in `get'
  /usr/local/lib/ruby/gems/2.7.0/gems/sawyer-0.8.2/lib/sawyer/agent.rb:94:in `call'
  /usr/local/lib/ruby/gems/2.7.0/gems/octokit-4.16.0/lib/octokit/connection.rb:156:in `request'
  /usr/local/lib/ruby/gems/2.7.0/gems/octokit-4.16.0/lib/octokit/connection.rb:19:in `get'
  /usr/local/lib/ruby/gems/2.7.0/gems/octokit-4.16.0/lib/octokit/client/pull_requests.rb:33:in `pull_request'
  /usr/local/lib/ruby/gems/2.7.0/gems/danger-6.3.0/lib/danger/request_sources/github/github.rb:117:in `fetch_details'
  /usr/local/lib/ruby/gems/2.7.0/gems/danger-6.3.0/lib/danger/danger_core/environment_manager.rb:52:in `fill_environment_vars'
  /usr/local/lib/ruby/gems/2.7.0/gems/danger-6.3.0/lib/danger/danger_core/dangerfile.rb:266:in `run'
  /usr/local/lib/ruby/gems/2.7.0/gems/danger-6.3.0/lib/danger/danger_core/executor.rb:29:in `run'
  /usr/local/lib/ruby/gems/2.7.0/gems/danger-6.3.0/lib/danger/commands/runner.rb:73:in `run'
  /usr/local/lib/ruby/gems/2.7.0/gems/claide-1.0.3/lib/claide/command.rb:334:in `run'
  /usr/local/lib/ruby/gems/2.7.0/gems/danger-6.3.0/bin/danger:5:in `<top (required)>'
  /usr/local/bin/danger:23:in `load'
  /usr/local/bin/danger:23:in `<top (required)>'
Failed to run bundle exec danger, error: exit status 1
```
## Your Environment

* Which CI are you running on?
Bitrise, Linux stack, Danger step v 2.0.0
* Are you running the latest version of Danger?
Yes
* What is your Dangerfile?

  ```ruby
  warn("Big PR") if git.lines_of_code > 500

  kotlin_detekt.skip_gradle_task = true
  kotlin_detekt.report_file = "build/reports/detekt/detekt.xml"
  kotlin_detekt.detekt(inline_mode: true)

  android_lint.skip_gradle_task = true
  android_lint.report_file = "build/reports/lint/lint-results.xml"
  android_lint.lint(inline_mode: true)

  github.dismiss_out_of_range_messages
  ```

Gemfile:
```
# frozen_string_literal: true

source "https://rubygems.org"

git_source(:github) {|repo_name| "https://github.com/#{repo_name}" }

gem 'danger-kotlin_detekt'
gem 'danger-android_lint'
```

**Comment by orta** [comment]
looks like the env vars didn't get hooked up right, I'd see where It got the PR number from

**Comment by ghaabor** [comment]
@orta @koral-- According to the GitHub API docs the URL should be `https://api.github.com/repos/DroidsOnRoids/<stripped>/pulls/189` and not `https://api.github.com/repos/DroidsOnRoids/<stripped>.git/pulls/189` but I might be mistaken.

**Comment by ghaabor** [comment]
I think this is already the fix for it: https://github.com/danger/danger/commit/e48ba2bd6d87b268597398ae3604fcc6340467b5

**Comment by ghaabor** [comment]
https://github.com/danger/danger/pull/1198

**Comment by angelolloqui** [comment]
Thanks for fixing! I spent a couple of hours trying to figure out why my private github repo integration  was not working, thinking that there was an issue with the api keys, permissions or alike! Moving to previous danger versions fixes the issue

**Comment by koral--** [comment]
I confirm that issue is fixed in 6.3.1.
